test/csuite/timestamp_abort/main.c

```diff
@@ -340,6 +363,218 @@ set_flush_tier_delay(WT_RAND_STATE *rnd)
+    while ((ret = cursor->next(cursor)) == 0) {
+        nfiles++;
+        testutil_check(cursor->get_key(cursor, &filename));
+        testutil_system("cp %s/%s %s/%s", WT_HOME_DIR, filename, backup_home, filename);
+    }
+    testutil_assert(ret == WT_NOTFOUND);
+
+    /* Cleanup */
+    testutil_check(cursor->close(cursor));
+    testutil_check(session->close(session, NULL));
+
+    /* Remember that the backup finished successfully. */
+    testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/done", backup_home));
+    testutil_assert((wfd = open(buf, O_WRONLY | O_CREAT, 0666)) >= 0);
+    testutil_assert(close(wfd) == 0);
+}
+
+/*
+ * backup_create_incremental --
+ *     Perform an incremental backup.
+ */
+static void
+backup_create_incremental(WT_CONNECTION *conn, uint32_t src_index, uint32_t index)
+{
+    WT_CURSOR *cursor, *file_cursor;
+    WT_SESSION *session;
+    ssize_t rdsize;
+    uint64_t offset, size, type;
+    int rfd, ret, wfd, nfiles, nranges;
+    char backup_home[PATH_MAX], src_backup_home[PATH_MAX];
+    char buf[4096];
+    char *filename;
+    bool first_range;
+    void *tmp;
+
+    nfiles = 0;
+    nranges = 0;
+
+    testutil_check(
+      __wt_snprintf(src_backup_home, sizeof(src_backup_home), "backup.%" PRIu32, src_index));
+    printf("Create backup: Type: incremental, Source ID: %" PRIu32 ", This ID: %" PRIu32 "\n",
+      src_index, index);
+
+    /* Prepare the directory. */
+    testutil_check(__wt_snprintf(backup_home, sizeof(backup_home), "backup.%" PRIu32, index));
+    testutil_system("rm -rf %s && mkdir %s", backup_home, backup_home);
+
+    /* Open the session. */
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Open the backup cursor to get the list of files to copy. */
+    testutil_check(__wt_snprintf(buf, sizeof(buf),
+      "incremental=(src_id=ID%" PRIu32 ",this_id=ID%" PRIu32 ")", src_index, index));
+    testutil_check(session->open_cursor(session, "backup:", NULL, buf, &cursor));
+
+    /* Process the files. */
+    while ((ret = cursor->next(cursor)) == 0) {
+        nfiles++;
+        testutil_check(cursor->get_key(cursor, &filename));
+
+        /* Process ranges within each file. */
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "incremental=(file=%s)", filename));
+        testutil_check(session->open_cursor(session, NULL, cursor, buf, &file_cursor));
+
+        first_range = true;
+        rfd = wfd = 0;
+        while ((ret = file_cursor->next(file_cursor)) == 0) {
+            testutil_check(file_cursor->get_key(file_cursor, &offset, &size, &type));
+            testutil_assert(type == WT_BACKUP_FILE || type == WT_BACKUP_RANGE);
+
+            if (type == WT_BACKUP_RANGE) {
+
+                /*
+                 * If this is the first range, copy the base file from the source backup. Then open
+                 * the source and target files to set up the rest of the copy.
+                 */
+                if (first_range) {
```

Comment:
Why are we copying with a RANGE type at all? If we have a range we want to just copy it. Using incremental should take a full backup first, which is what you do in `backup_create_full`. So your target files should be there already.

Reply:
I think a good comment explaining what is going on (and maybe a few smaller comments in the code elsewhere referencing some aspect of the strategy) will be sufficient. Thanks for you explanation.